web/concrete/src/Service/Configuration/HTTP/NginxGenerator.php

```diff
@@ -0,0 +1,119 @@
+<?php
+namespace Concrete\Core\Service\Configuration\HTTP;
+
+use Concrete\Core\Service\Configuration\GeneratorInterface;
+
+class NginxGenerator implements GeneratorInterface
+{
+    /**
+     * @var Application
+     */
+    protected $app;
+
+    /**
+     * @var array
+     */
+    protected $rules;
+
+    /**
+     * @var array
+     */
+    protected $enabledRules;
+
+    /**
+     * Initializes the instance.
+     *
+     * @param Application $app
+     */
+    public function __construct(Application $app)
```

Comment:
Interfaces are always good to have since PHP is single inheritance. Classes can only have a single parent class, but they can have infinite interfaces or traits. With interfaces, someone can make a custom object that includes real types from both concrete5 and another vendor enabling extremely powerful interoperability. Imagine:

```php
class CustomUser implements 
    LaravelUser, 
    DrupalUser, 
    Concrete5User
{
   ...
}
```
That's interoperability at it's finest and only possible if we declare interfaces :beers: 


Thinking about this specific case, you probably wouldn't need the base generator to be abstract since the only thing that is different is the constructor. If you just had a basic generator implementation, people could do:

```php
$generator = new \Concrete\Core\Service\Configuration\HTTP\Generator();
$generator->addRule('pretty_urls', ...);
```

Without having to create a new custom class. Then you can just make the ApacheGenerator and the NginxGenerator extend with only the constructor.